gnome-image-installer/pages/diskimage/gis-diskimage-page.c

```diff
@@ -390,13 +396,13 @@ gis_diskimage_page_add_live_image (
 
   if (file_exists (live_device_path, NULL))
     {
-      add_image (store, endless_img_path, live_device_path, live_sig);
+      add_image (store, endless_path, live_device_path, live_sig);
     }
   else
     {
       g_print ("can't find image device %s; will use %s directly\n",
-               live_device_path, endless_img_path);
-      add_image (store, endless_img_path, NULL, live_sig);
+               live_device_path, endless_path);
+      add_image (store, endless_path, NULL, live_sig);
```

Comment:
This fallback path doesn't make sense when `endless_path` is `.../endless.squash`, right? The signature won't match, and writing the squash image directly to disk wouldn't boot. I think it would be better to skip this path entirely in that case.